### ISSUE TITLE
Spec refactor

### ISSUE THREAD

**Issue by webberwang** [codelab-ai/codelab.ai#973]
<img width="462" alt="create-lambda_i_spec_ts_—_codelab" src="https://user-images.githubusercontent.com/26264925/126905572-87cfe0cb-a8d4-446c-8df1-2b148c6dba5d.png">

Follow `create-lambda.i.spec.ts` & refactor our specs. Using the `domainRequest` method, we don't need to create an individual helper method per model we're testing. This also extracts the variable to the spec file that is calling the function.

There are also better error logging, and improves on how we test expected errors.

This should be a quick refactor before we move on to some other tasks.

---

7.26 UPDATE

<img width="502" alt="delete-lambda_i_spec_ts_—_codelab" src="https://user-images.githubusercontent.com/26264925/126923972-83e75850-5350-43cc-992c-f05c70c846b8.png">

We should also assert that the returned ID is defined in cases where we don't know what the id will be. If we know the ID like update, we should assert it

---

Re-use create data, this will be imported for update/delete etc

<img width="326" alt="LibraryService_i_spec_ts_—_codelab" src="https://user-images.githubusercontent.com/26264925/126924717-2e7c66e8-8a99-4c46-9ec8-505743d12c1e.png">

Place all specs related under `test` folder inside use case

---

7.28 UPDATE

<img width="555" alt="delete-lambda_i_spec_ts_—_codelab" src="https://user-images.githubusercontent.com/26264925/127293607-032df4e1-fd19-49c1-8b64-61f90b30f709.png">

I noticed in a flaw for our delete specs. We were checking if the results from the delete service matched the id input, but this doesn't check if the item is being deleted. A flaw in the delete service returned the id even it wasn't being deleted.

Instead, we perform a get request to check for it in the database.

---

In other words, we can't rely on returned item for verifying that data is persisted. For create/update/delete, we need to do a get to verify.

**Comment by angelwong0963** [comment]
Let me fix it.

**Comment by angelwong0963** [comment]
I found another issue that `domainRequest()` doesn't check the `expectedError` parameter correctly.
![image](https://user-images.githubusercontent.com/85784065/127306251-5f89a16b-6834-45db-a249-232319d97aec.png)

` nx run modules-lambda-api:test --runInBand`
![image](https://user-images.githubusercontent.com/85784065/127306363-2bada508-433f-4174-af46-d0c19e1d8fd8.png)

**Comment by webberwang** [comment]
> I found another issue that `domainRequest()` doesn't check the `expectedError` parameter correctly.
> ![image](https://user-images.githubusercontent.com/85784065/127306251-5f89a16b-6834-45db-a249-232319d97aec.png)
> 
> ` nx run modules-lambda-api:test --runInBand`
> ![image](https://user-images.githubusercontent.com/85784065/127306363-2bada508-433f-4174-af46-d0c19e1d8fd8.png)

Oh good catch, didn’t see that

**Comment by angelwong0963** [comment]
via Hubstaff
User: Angel Wong

Project: codelab.ai - https://app.hubstaff.com/projects/1192287
Date Range: 07/26/21 - 07/30/21
Work session total: 8:28:41
Billable: Yes

Grand total: 8:28:41
